### PR TITLE
Mark experimental-cluster-signing-duration deprecated, add --cluster-signing-duration

### PR DESCRIPTION
**Pull request by liggitt**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Graduates the kube-controller-manager `experimental-cluster-signing-duration` flag to `cluster-signing-duration`. Part of https://github.com/kubernetes/enhancements/issues/1513.

**Does this PR introduce a user-facing change?**:
```release-note
kube-controller-manager: the `--experimental-cluster-signing-duration` flag is marked as deprecated for removal in v1.22, and is replaced with `--cluster-signing-duration`.
```

/cc @mikedanese 
/sig auth

### CODE REVIEW

File: cmd/kube-controller-manager/app/options/csrsigningcontroller.go [hunk 1]
```diff
@@ -45,7 +45,9 @@ func (o *CSRSigningControllerOptions) AddFlags(fs *pflag.FlagSet) {
 
 	fs.StringVar(&o.ClusterSigningCertFile, "cluster-signing-cert-file", o.ClusterSigningCertFile, "Filename containing a PEM-encoded X509 CA certificate used to issue cluster-scoped certificates")
 	fs.StringVar(&o.ClusterSigningKeyFile, "cluster-signing-key-file", o.ClusterSigningKeyFile, "Filename containing a PEM-encoded RSA or ECDSA private key used to sign cluster-scoped certificates")
+	fs.DurationVar(&o.ClusterSigningDuration.Duration, "cluster-signing-duration", o.ClusterSigningDuration.Duration, "The length of duration signed certificates will be given.")
 	fs.DurationVar(&o.ClusterSigningDuration.Duration, "experimental-cluster-signing-duration", o.ClusterSigningDuration.Duration, "The length of duration signed certificates will be given.")
+	fs.MarkDeprecated("experimental-cluster-signing-duration", "use --cluster-signing-duration")
```
Comment (munnerz) [inline cmd/kube-controller-manager/app/options/csrsigningcontroller.go:50]: What happens if both flags are specified?

Comment (liggitt) [reply cmd/kube-controller-manager/app/options/csrsigningcontroller.go:50]: they address the same variable, the last one specified wins, and use of the deprecated one prints a warning

**Comment by enj** [comment]
/lgtm

This seems like a step in the right direction.  I have some concerns about the coarseness of `--cluster-signing-duration` but will open issues to start the discussion.

**Pull request merged**